src/vSphereVirtualMachineCollector.class.inc.php

```diff
@@ -307,7 +307,7 @@ static protected function DoCollectVMInfo($aFarms, $oVirtualMachine, $aVLANs, $i
 				$sGuestIP = '';
 			} else {
 				if (!$bCollectIPv6Addresses) {
-					$sGuestIP = (strpos($sGuestIP, ':') !== false) ? '' : $sGuestIP;
+					$sGuestIP = filter_var($sGuestIP, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6) ?: '';
```

Comment:
Since we don't want to consider IPv6s here and leave an empty string if it's one, I think the correct syntax is:
$sGuestIP = filter_var($sGuestIP, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4) ?: '';

Reply:
```suggestion
					$sGuestIP = filter_var($sGuestIP, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4) ?: '';
```